src/main/java/org/graylog/integrations/ipfix/codecs/IpfixCodec.java

```diff
@@ -82,18 +91,54 @@
     protected IpfixCodec(@Assisted Configuration configuration, IpfixAggregator ipfixAggregator) throws IOException {
         super(configuration);
         this.ipfixAggregator = ipfixAggregator;
-        // Standard IPFIX definition file is loaded from the classpath and is mandatory. We don't check for null or empty.
+
         final URL standardIPFixDefTemplate = Resources.getResource(IpfixCodec.class, IPFIX_STANDARD_DEFINITION);
-        final String ipFixCustomDefPath = (String) configuration.getSource().get(CK_IPFIX_DEFINITION_PATH);
-        if (ipFixCustomDefPath == null || ipFixCustomDefPath.trim().isEmpty()) {
+        final List<String> customDefFilePathList = configuration.getList(CK_IPFIX_DEFINITION_PATH);
+        List<URL> filePaths = new ArrayList<>();
+
+        if (customDefFilePathList == null || customDefFilePathList.isEmpty()) {
             infoElementDefs = new InformationElementDefinitions(standardIPFixDefTemplate);
         } else {
-            final URL customIPFixDefURL = Paths.get(ipFixCustomDefPath).toUri().toURL();
-            infoElementDefs = new InformationElementDefinitions(standardIPFixDefTemplate, customIPFixDefURL);
+            errOnInvalidFilePath(customDefFilePathList);
+            filePaths.add(standardIPFixDefTemplate);
+            for (String filePath : customDefFilePathList) {
+                URL customDefURL = url(filePath.trim());
+                filePaths.add(customDefURL);
+            }
+            URL[] urls = array(filePaths);
+            infoElementDefs = new InformationElementDefinitions(urls);
         }
         this.parser = new IpfixParser(this.infoElementDefs);
     }
 
+    URL url(String s) throws MalformedURLException {
+        return Paths.get(s).toUri().toURL();
+    }
+
+    URL[] array(List<URL> urls) {
+        URL[] urlArray = new URL[urls.size()];
+        return urls.toArray(urlArray);
+    }
+
+    void errOnInvalidFilePath(List<String> customDefFilePathList) throws IpfixException {
+        for (String filePath : customDefFilePathList) {
+            File file = new File(filePath.trim());
+            validateFilePath(file);
+        }
+    }
+
+    public void validateFilePath(File customDefFile) throws IpfixException {
+        if (customDefFile.isDirectory()) {
+            String message = "The specified path is a folder. Please specify the full path to the file.";
+            LOG.debug(message);
+            throw new IpfixException("An error occurred due to the following error [" + message + "]");
+        } else if (!customDefFile.exists()) {
+            String message = "The specified file [%s] does not exist.";
```

Comment:
The format string (`%s`) in the message will not be substituted because we don't use `String.format()` here.

Reply:
This has been removed